src/kernel/type_checker.cpp

```diff
@@ -445,8 +446,10 @@ void check_no_metavar(environment const & env, name const & n, expr const & e, b
 }
 
 static void check_no_local(environment const & env, expr const & e) {
-    if (has_local(e))
+    if (has_local(e)) {
+        std::cout << "has locals: " << e << std::endl;
```

Comment:
Yes, my mistake. I thought I had re-checked out the original but I see I did not. Obviously the `cout` was a temporary hack, but it is annoying that the exception does not print the expression. It would be nice to have the message: `Expression <expr> contains the following local constants: <local> ... <local>`. How would you recommend I implement this? With tracing?